pkgs/by-name/ty/typst/typst-packages.nix

```diff
@@ -0,0 +1,39 @@
+{
+  lib,
+  callPackage,
+}:
+
+lib.makeExtensible (
+  final:
+  lib.recurseIntoAttrs (
+    lib.mapAttrs (
+      name: packageSpec:
+      callPackage (
+        {
+          lib,
+          buildTypstPackage,
+          fetchurl,
+        }:
+        buildTypstPackage {
+          inherit (packageSpec) pname version;
+
+          src = fetchurl {
+            inherit (packageSpec) url hash;
+          };
+
+          sourceRoot = ".";
+
+          typstDeps = builtins.filter (x: x != null) (
+            lib.map (d: (lib.attrsets.attrByPath [ d ] null final)) packageSpec.typstDeps
+          );
+
+          meta = {
+            inherit (packageSpec) description;
+            maintainers = with lib.maintainers; [ cherrypiejam ];
+            license = lib.map (lib.flip lib.getAttr lib.licensesSpdx) packageSpec.license;
+          } // (if packageSpec ? "homepage" then { inherit (packageSpec) homepage; } else { });
+        }
+      ) { }
+    ) (lib.importTOML ./typst-packages-from-universe.toml)
```

Comment:
Might be a good idea to check if it evaluates with [minimal supported Nix version](https://github.com/NixOS/nixpkgs/blob/master/lib/minver.nix). While the builtin is [supported since 2.1](https://github.com/NixOS/nix/blob/e4bda20918ad2af690c2e938211a7d362548e403/doc/manual/source/release-notes/rl-2.1.md?plain=1#L21), the parser [changed in 2.6](https://github.com/NixOS/nix/blob/e4bda20918ad2af690c2e938211a7d362548e403/doc/manual/source/release-notes/rl-2.6.md?plain=1#L6).